cli/lsp/diagnostics.rs

```diff
@@ -1005,17 +996,43 @@ fn diagnose_dependency(
   diagnose_resolution(
     diagnostics,
     snapshot,
-    &dependency.maybe_code,
-    dependency.is_dynamic,
-    dependency.maybe_assert_type.as_deref(),
-  );
-  diagnose_resolution(
-    diagnostics,
-    snapshot,
-    &dependency.maybe_type,
+    if dependency.maybe_code.is_none() {
+      &dependency.maybe_type
+    } else {
+      &dependency.maybe_code
+    },
     dependency.is_dynamic,
     dependency.maybe_assert_type.as_deref(),
+    dependency
+      .imports
+      .iter()
+      .map(|i| documents::to_lsp_range(&i.range))
+      .collect(),
   );
+  // TODO(nayeemrmn): This is a crude way of detecting `@deno-types` which has
+  // a different specifier and therefore needs a separate call to
+  // `diagnose_resolution()`. It would be much cleaner if that were modelled as
+  // a separate dependency: https://github.com/denoland/deno_graph/issues/247.
+  if !dependency.maybe_type.is_none()
+    && !dependency
+      .imports
+      .iter()
+      .any(|i| dependency.maybe_type.includes(&i.range.start).is_some())
+  {
+    let range = match &dependency.maybe_type {
+      Resolution::Ok(resolved) => documents::to_lsp_range(&resolved.range),
+      Resolution::Err(error) => documents::to_lsp_range(error.range()),
+      Resolution::None => unreachable!(),
```

Comment:
Are we sure about this? Maybe just have it ignore this scenario instead?

Reply:
It's gated in the containing if statement